### DISCUSSION TITLE
Redshift's REMOVE DUPLICATES not available on MERGE INTO

### DISCUSSION THREAD

**Discussion by xmariachi** [psycopg/psycopg2#1633] [General]
Using pyscopg 2.9.5. 
Redshift introduced the [MERGE](https://docs.aws.amazon.com/redshift/latest/dg/r_MERGE.html) syntax just recently (April 2023). 
**Question - Is it supported at all ?** 

It has two optional clauses: 
- WHEN MATCHED ... WHEN NOT MATCHED 
- REMOVE DUPLICATES

I am able to use the former sql syntax, but the latter does not seem to be supported: 
```
psycopg2.errors.SyntaxError: syntax error at or near "REMOVE"
```

The tables are identical.
Query run: 
```
 MERGE INTO a
  USING b
  ON a.id = b.id
  REMOVE DUPLICATES;
```

Thanks.

**Comment by dvarrazzo** [comment]
The error comes from the server. Psycopg just send the commands to the server, it doesn't give them any meaning. So you have a genuine error in your sql.

**Reply by xmariachi** [reply]
Hmmm I see.... I will review, thank you
